packages/flutter/test/material/app_test.dart

```diff
@@ -307,16 +307,27 @@ void main() {
     expect(log, <String>['onGenerateRoute /', 'onUnknownRoute /']);
   });
 
-  testWidgets('reacts to changing the text scale', (WidgetTester tester) async {
+  testWidgets('Reacts to changing the text scale', (WidgetTester tester) async {
     double textScaleFactor;
-    ui.TestingHooks.updateTextScaleFactor(1.5);
     await tester.pumpWidget(new MaterialApp(
       home: new Builder(builder:(BuildContext context) {
         textScaleFactor = MediaQuery.of(context).textScaleFactor;
         return new Container();
       }),
     ));
+    expect(textScaleFactor, isNotNull);
+    expect(textScaleFactor, equals(1.0));
+    textScaleFactor = null;
 
+    // Now simulate a message from the engine that the text scale factor has changed.
+    ui.TestingHooks.updateTextScaleFactor(1.5);
+
+    await tester.pumpWidget(new MaterialApp(
+      home: new Builder(builder:(BuildContext context) {
+        textScaleFactor = MediaQuery.of(context).textScaleFactor;
+        return new Container();
+      }),
+    ));
```

Comment:
What I meant was a test that actually verified that the Text of, say, an AppBar widget, or of a FlatButton, actually grew.

Reply:
Well, but shouldn't that be tested by a different test?  Seems like as long as the textScaleFactor changes in the media query, then this code has done its job.  Looks like the MediaQuery and Text tests don't really do that kind of testing, however.  I'll see if I can add something there.

Reply:
I don't mind how many tests you write or how you split it up. :-)

Until now we've not done anything with textScaleFactor, so it's completely untested. We should verify that widgets get bigger when the font is scaled. We can test this as one end-to-end test or many tests that check each link in the chain, either way is fine by me.

Reply:
So, I tried adding a test to text_test.dart, and I'm perplexed.  Why does this fail at the expect for `largeSize.height`?  It thinks that text with scale factor 3.0 is the same size as text with a scale of 1.0 (14 px).

```dart
  testWidgets('Text respects textScaleFactor', (WidgetTester tester) async {
    await tester.pumpWidget(const Center(
        child: const Text('Hello', textScaleFactor: 3.0, textDirection: TextDirection.ltr)
    ));

    final RichText text = tester.firstWidget(find.byType(RichText));
    expect(text.textScaleFactor, 3.0);
    final Size largeSize = tester.getSize(find.byType(RichText));
    expect(largeSize.height, equals(3.0 * 14.0));
  });
```
Is this some issue with the Linux engine vs the regular engine?  Because text is clearly scaled on a device.  I just can't see how paragraph layout could work on the tests if it can't reliably determine the text size.

Note that this doesn't even involve MediaQuery, so it doesn't have anything to do with that.

Reply:
Turns out that this is a bug in the way the default text size is handled. I fixed it and added some more tests.